expression/builtin_math.go

```diff
@@ -1028,10 +1028,8 @@ func (b *builtinRandWithSeedSig) evalReal(row chunk.Row) (float64, bool, error)
 	// b.args[0] is promised to be a non-constant(such as a column name) in
 	// builtinRandWithSeedSig, the seed is initialized with the value for each
 	// invocation of RAND().
-	var randGen *rand.Rand
-	if isNull {
-		randGen = rand.New(rand.NewSource(time.Now().UnixNano()))
-	} else {
+	randGen := rand.New(rand.NewSource(time.Now().UnixNano()))
+	if !isNull {
 		randGen = rand.New(rand.NewSource(seed))
```

Comment:
Please keep the `else` branch here then we can avoid creating an unnecessary rander if there is a seed.

Reply:
I had modified.